app/models/foreman_tasks/task.rb

```diff
@@ -83,18 +86,28 @@ def self.search_by_generic_resource(key, operator, value)
     end
 
     def self.search_by_owner(key, operator, value)
+      return { :conditions => '0 = 1' } if value == 'current_user' && User.current.nil?
+
       key_name = self.connection.quote_column_name(key.sub(/^.*\./,''))
-      joins = <<-JOINS
+      joins = <<-JOIN_OWNER
```

Comment:
if `<<-SQL` is used instead it will allow certain editors (Rubymine) to highlight the SQL syntax.